### PR TITLE
[Backend] - Cancel Order API

### PR DESCRIPTION
**Pull request by TreshAnn**
## Ticket/s

- [Cancel Order](https://trello.com/c/vGdhVyRF)

## Changes

- Add Cancel Order API

## Screenshot

image

## How to test?

Note: Must have user data and order data before testing
- Open Postman
- Log in as a user to obtain the access token
- click import tab, paste cURL
```
curl --location --request POST 'http://localhost:3001/api/order/cancel/[Insert Order _id here]' \
--header '[Insert Access Token here]'
```
- Copy the obtained token to the Authorization Header

### CODE REVIEW

**Review by SGVGen — APPROVED** [review]
LGTM!

File: apps/api/src/order/order.service.ts [hunk 1]
```diff
@@ -112,9 +109,47 @@ export class OrderService {
...
+    if (
+      order.status === 'shipped' ||
+      order.status === 'processing' ||
+      order.status === 'delivered'
+    ) {
+      throw new BadRequestException(
+        'Cannot cancel an order with status: ' + order.status,
+      );
+    }
+
```
Comment (brentmillen) [inline apps/api/src/order/order.service.ts:137]: utilize request payload to validate cancellation of own orders only

File: apps/api/src/order/schemas/order.schema.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export class Order {
   @Prop({ default: Date.now })
   date: Date;
 
-  @Prop({ enum: ['ordered', 'processing', 'shipped', 'delivered'] })
+  @Prop({ enum: ['ordered', 'processing', 'shipped', 'delivered', 'Canceled'] })
```
Comment (brentmillen) [inline apps/api/src/order/schemas/order.schema.ts]: same casing for all values

File: apps/api/src/products/products.service.ts [hunk 3]
```diff
@@ -137,4 +137,8 @@ export class ProductsService {
 
     return product;
   }
+  async findProductByName(productName: string): Promise<Product> {
+    const product = await this.productModel.findOne({ productName });
```
Comment (brentmillen) [inline apps/api/src/products/products.service.ts]: add exception handler; is it also possible to use the objectId instead?

**Review by sandrex-SGV — APPROVED** [review]
LGTM

**Review by brentmillen — APPROVED** [review]
LGTM

**Review by SGVGen — APPROVED** [review]
LGTM!

**Pull request merged**